introduce VMAnomalyConfig (#1472)
VictoriaMetrics/operator: docs/CHANGELOG.md

**Update note 2**: removed `operator_prometheus_converter_watch_events_total` metric since migration of Prometheus object watchers to controllers made this counter obsolete.
**Update note 3**: made `controller.prometheusCRD.resyncPeriod` command line flag noop, which was relevant to Prometheus object watchers.
**Update note 4**: `-eula` flag is not set by default anymore for VMBackup and VMRestore. To avoid VMCluster/VMSingle rollouts set `spec.vmstorage.vmBackup.acceptEula: true` for VMCluster and `spec.vmBackup.acceptEula: true"` for VMSingle and replace it with `spec.license` during VMSingle/VMCluster upgrade.

> **Update note 5**: due to introduction of new flags and sidecar VMAnomaly will be recreated.

Reviewer: ```suggestion
**Update note 5**: new flags and config-reloader sidecar was introduced, so VMAnomaly pods will be recreated.
```

> **Update note 5**: new flags and config-reloader sidecar was introduced, so VMAnomaly pods will be recreated.

Author: actually PR introduces no config-reloader, using --watch flag instead, Nik was against it
